File: projetpi/art/src/main/java/models/livraison.java
package models;

import java.util.Date;

public class livraison {

        private  int id_livraison;
        private String date_livraison;
        private String adresse_livraison;
        private String status_livraison;
        private Float frais_livraison;

    public int getId_commande() {
        return id_commande;
    }

    public void setId_commande(int id_commande) {
        this.id_commande = id_commande;
    }

    private int id_commande;

    public int getId_livreur() {
        return id_livreur;
    }

    public void setId_livreur(int id_livreur) {
        this.id_livreur = id_livreur;
    }

    private int id_livreur;

        public livraison() {

        }


        public livraison(String date_livraison, String adresse_livraison, String status_livraison, Float frais_livraison, int id_livreur, int id_commande) {

            this.date_livraison = date_livraison;
            this.adresse_livraison = adresse_livraison;
            this.status_livraison = status_livraison;
            this.frais_livraison = frais_livraison;
            this.id_livreur = id_livreur;
            this.id_commande = id_commande;
        }

    public livraison(int id_livraison,String date_livraison, String adresse_livraison, String status_livraison, Float frais_livraison, int id_livreur, int id_commande) {
          this.id_livraison=id_livraison;
        this.date_livraison = date_livraison;
        this.adresse_livraison = adresse_livraison;
        this.status_livraison = status_livraison;
        this.frais_livraison = frais_livraison;
        this.id_livreur = id_livreur;
        this.id_commande = id_commande;
    }

        // Getters and Setters

        public int getID() {
            return id_livraison;
        }

        public void setID(int id_livraison) {
            this.id_livraison=id_livraison;
        }



        public String getDate() {
            return date_livraison;
        }

        public void setDate(String date_livraison) {
            this.date_livraison = date_livraison;
        }

        public String getAdresse() {
            return adresse_livraison;
        }

        public void setAdresse(String adresse_livraison) {
            this.adresse_livraison = adresse_livraison;
        }

        public String getStatus() {
            return status_livraison;
        }

        public void setStatus(String status_livraison) {
            this.status_livraison = status_livraison;
        }


        public Float getFrais() {
        return frais_livraison;
        }

        public void setFrais(Float frais_livraison) {
        this.frais_livraison = frais_livraison;
        }






    }

File: src/Entity/Livraison.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Livraison
 *
 * @ORM\Table(name="livraison", indexes={@ORM\Index(name="fk_livraison_commande", columns={"id_commande"}), @ORM\Index(name="fk_livraison_livreur", columns={"id_livreur"})})
 * @ORM\Entity
 */
class Livraison
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_livraison", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idLivraison;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_livraison", type="date", nullable=false)
     */
    private $dateLivraison;

    /**
     * @var string
     *
     * @ORM\Column(name="adresse_livraison", type="string", length=50, nullable=false)
     */
    private $adresseLivraison;

    /**
     * @var string
     *
     * @ORM\Column(name="status_livraison", type="string", length=20, nullable=false)
     */
    private $statusLivraison;

    /**
     * @var float
     *
     * @ORM\Column(name="frais_livraison", type="float", precision=10, scale=0, nullable=false)
     */
    private $fraisLivraison;

    /**
     * @var \Commande
     *
     * @ORM\ManyToOne(targetEntity="Commande")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_commande", referencedColumnName="id_commande")
     * })
     */
    private $idCommande;

    /**
     * @var \Livreur
     *
     * @ORM\ManyToOne(targetEntity="Livreur")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_livreur", referencedColumnName="id_livreur")
     * })
     */
    private $idLivreur;

    public function getIdLivraison(): ?int
    {
        return $this->idLivraison;
    }

    public function getDateLivraison(): ?\DateTimeInterface
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(\DateTimeInterface $dateLivraison): self
    {
        $this->dateLivraison = $dateLivraison;

        return $this;
    }

    public function getAdresseLivraison(): ?string
    {
        return $this->adresseLivraison;
    }

    public function setAdresseLivraison(string $adresseLivraison): self
    {
        $this->adresseLivraison = $adresseLivraison;

        return $this;
    }

    public function getStatusLivraison(): ?string
    {
        return $this->statusLivraison;
    }

    public function setStatusLivraison(string $statusLivraison): self
    {
        $this->statusLivraison = $statusLivraison;

        return $this;
    }

    public function getFraisLivraison(): ?float
    {
        return $this->fraisLivraison;
    }

    public function setFraisLivraison(float $fraisLivraison): self
    {
        $this->fraisLivraison = $fraisLivraison;

        return $this;
    }

    public function getIdCommande(): ?Commande
    {
        return $this->idCommande;
    }

    public function setIdCommande(?Commande $idCommande): self
    {
        $this->idCommande = $idCommande;

        return $this;
    }

    public function getIdLivreur(): ?Livreur
    {
        return $this->idLivreur;
    }

    public function setIdLivreur(?Livreur $idLivreur): self
    {
        $this->idLivreur = $idLivreur;

        return $this;
    }


}
